### PR TITLE
Update version docs (1.1.0)

### LINKED ISSUE #2893: SSL_get_version will return a version before handshake.
Spotted originally when working with Python, as [Python issue 29781](https://bugs.python.org/issue29781).

`SSL_get_version` is documented as follows: 

```
NAME
       SSL_get_version - get the protocol version of a connection.

SYNOPSIS
        #include <openssl/ssl.h>

        const char *SSL_get_version(const SSL *ssl);

DESCRIPTION
       SSL_get_version() returns the name of the protocol used for the
       connection ssl.

RETURN VALUES
       The following strings can be returned:

       SSLv2
           The connection uses the SSLv2 protocol.

       SSLv3
           The connection uses the SSLv3 protocol.

       TLSv1
           The connection uses the TLSv1.0 protocol.

       TLSv1.1
           The connection uses the TLSv1.1 protocol.

       TLSv1.2
           The connection uses the TLSv1.2 protocol.

       unknown
           This indicates that no version has been set (no connection
           established).
```

Note that the man page says that "unknown" will be returned when no connection is established. However, this is not the actual behaviour. For example, consider the following test code:

```c
#include <stdio.h>
#include <openssl/bio.h>
#include <openssl/ssl.h>

void init_openssl(void) { 
    SSL_library_init();
    SSL_load_error_strings();
    ERR_load_BIO_strings();
    OpenSSL_add_all_algorithms();
}

SSL_CTX *create_context() {
    const SSL_METHOD *method;
    SSL_CTX *ctx;

    method = SSLv23_server_method();

    ctx = SSL_CTX_new(method);
    if (!ctx) {
	printf("Unable to create context!\n");
	exit(1);
    }

    return ctx;
}

int main(int argc, char **argv) {
    SSL_CTX *ctx;
    SSL *ssl;
    BIO *in_bio;
    BIO *out_bio;
    const char *version;

    init_openssl();
    ctx = create_context();
    ssl = SSL_new(ctx);
    in_bio = BIO_new(BIO_s_mem());
    out_bio = BIO_new(BIO_s_mem());

    if ((in_bio == NULL) || (out_bio == NULL)) {
        printf("Unable to allocate BIOs\n");
        exit(2);
    }

    SSL_set_bio(ssl, in_bio, out_bio);
    SSL_set_connect_state(ssl);

    version = SSL_get_version(ssl);
    printf("Version: %s\n", version);

    SSL_free(ssl);
    SSL_CTX_free(ctx);
    return 0;
}
```

This test code prints `Version: TLSv1.2`. This is despite the fact that no connection is established, and that as a result there is no TLS version in use right now.

I've reproduced this bug against OpenSSL 1.0.2, but I have no reason to believe it was not present in earlier versions, or in later ones.

### PR DESCRIPTION
**Pull request by mattcaswell**
Make it clear that you should not attempt to get the version before the
first handshake is complete.

Fixes #2893

This is the 1.1.0 version of #6013.

##### Checklist

- [ ] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

File: doc/ssl/SSL_get_version.pod [hunk 1]
```diff
@@ -17,6 +17,10 @@ SSL_get_version, SSL_is_dtls - get the protocol information of a connection
 SSL_get_version() returns the name of the protocol used for the
 connection B<ssl>.
 
+The above function should only be called after the intial handshake has been
+completed. Prior to that the results returned from this function may be
+unreliable.
```
Comment (levitte) [inline doc/ssl/SSL_get_version.pod]: Same comment here as for the [1.0.2 PR](https://github.com/openssl/openssl/pull/6015#pullrequestreview-113609883)

Comment (mattcaswell) [reply doc/ssl/SSL_get_version.pod]: Fixed

**Comment by mattcaswell** [comment]
Updated with same wording change as #6013.

**Comment by mattcaswell** [comment]
Pushed. Thanks.

**Pull request closed without merging**